pkg/report/decompile.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2021 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package report
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/google/syzkaller/pkg/osutil"
+	"github.com/google/syzkaller/sys/targets"
+)
+
+type DecompilerFlag string
+
+// Extra flags that control the flow of decompilation.
+const (
+	FlagForceArmThumbMode DecompilerFlag = "force_arm_thumb"
+)
+
+const objdumpCallTimeout = 10 * time.Second
+
+type DecompiledOpcode struct {
+	Offset  int
+	Len     int
+	Command string
+	IsBad   bool
+	// Yes, the field below duplicates information and can be inferred from the other fields,
+	// but at this point it does not seem to be reasonable to parse a string and then build
+	// exactly that string from its parts.
+	FullDescription string
+}
+
+// Decompiles a byte array with opcodes into human-readable descriptions.
+// Target must specify the environment from which the opcodes were taken.
+func DecompileOpcodes(raw []byte, flags []DecompilerFlag, target *targets.Target) ([]DecompiledOpcode, error) {
+	args, err := objdumpBuildArgs(flags, target)
+	if err != nil {
+		return nil, err
+	}
+
+	outLines, err := objdumpExecutor(raw, args, target)
+	if err != nil {
+		return nil, err
+	}
+
+	list := objdumpParseOutput(outLines)
+	if len(list) == 0 && len(raw) > 0 {
+		return nil, fmt.Errorf("no instructions found while the total size is %v bytes", len(raw))
+	}
+	return list, nil
+}
+
+func objdumpExecutor(raw []byte, args []string, target *targets.Target) ([]string, error) {
+	file, err := ioutil.TempFile("", "")
```

Comment:
Use some recognizable prefix for temp files, we commonly use "syz-something". When some temp files pile up in /tmp due to a bug or something, it useful to be able to easily track what created them.